app/src/main/java/com/gusta/wakemehome/geofencing/GeofenceManager.java

```diff
@@ -276,102 +277,15 @@ private void handleError(Exception e) {
-                            mContextWrapper.startActivity(intent);
-                        }
-                    });
-        } else {    // The context wrapper is not an activity - a notification needs to be used
-            NotificationUtils.sendNotification(mContextWrapper,
-                    mContextWrapper.getString(R.string.geofence_not_available_title),
-                    mContextWrapper.getString(R.string.geofence_not_available_text), intent);
-        }
-    }
-
-    /**
-     * Shows a {@link Snackbar} using {@code text}.
-     *
-     * @param text The Snackbar text.
-     */
-    private void showSnackbar(final String text) {
-        // Abort if the context wrapper is not an activity (can't show messages to the user)
-        if (!(mContextWrapper instanceof Activity)) return;
-        Activity activity = (Activity) mContextWrapper;
-
-        // Show the message to the user
-        View container = activity.findViewById(android.R.id.content);
-        if (container != null) {
-            Snackbar.make(container, text, Snackbar.LENGTH_LONG).show();
-        }
-    }
-
-    /**
-     * Shows a {@link Snackbar}.
-     *
-     * @param mainTextStringId The id for the string resource for the Snackbar text.
-     * @param actionStringId   The text of the action item.
-     * @param listener         The listener associated with the Snackbar action.
-     */
-    private void showSnackbar(final int mainTextStringId, final int actionStringId,
-                              View.OnClickListener listener) {
-        // Abort if the context wrapper is not an activity (can't show messages to the user)
-        if (!(mContextWrapper instanceof Activity)) return;
-        Activity activity = (Activity) mContextWrapper;
-
-        // Show the message to the user
-        Snackbar.make(
-                activity.findViewById(android.R.id.content),
-                mContextWrapper.getString(mainTextStringId),
-                Snackbar.LENGTH_INDEFINITE)
-                .setAction(mContextWrapper.getString(actionStringId), listener).show();
-    }
-
-    /**
-     * Ask the user for permission.
-     */
-    private void requestPermissions() {
-        // Abort if the context wrapper is not an activity (can't show messages to the user)
-        if (!(mContextWrapper instanceof Activity)) return;
-
-        // Request permissions from the user
-        final Activity activity = (Activity) mContextWrapper;
-        boolean shouldProvideRationale =
-                ActivityCompat.shouldShowRequestPermissionRationale(activity,
-                        Manifest.permission.ACCESS_FINE_LOCATION);
-
-        // Provide an additional rationale to the user. This would happen if the user denied the
-        // request previously, but didn't check the "Don't ask again" checkbox.
-        if (shouldProvideRationale) {
-            Log.i(TAG, "Displaying permission rationale to provide additional context.");
```

Comment:
I think it should be added to `strings.xml`

Reply:
This is a debug log - if we start adding them to "strings.xml" it will just make it a huge file we can't find anything at (Plus, while debugging we would usually add/removes logs like that - it will just irritate us to put them in strings.xml)